indico/migrations/versions/20171211_0811_03dca79b401c_hidden_folders_support.py

```diff
@@ -0,0 +1,29 @@
+"""Hidden folders support
+
+Revision ID: 03dca79b401c
+Revises: 566d5de4e0e5
+Create Date: 2017-12-11 08:11:23.673487
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = '03dca79b401c'
+down_revision = '566d5de4e0e5'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column('folders', sa.Column('is_hidden', sa.Boolean(), nullable=False, server_default='false'),
+                  schema='attachments')
+    op.alter_column('folders', 'is_hidden', server_default=None, schema='attachments')
+    op.create_check_constraint('ck_folders_exclusive_is_hidden_is_always_visible', 'folders',
```

Comment:
Not sure if a DB constraint is really important here - there is no harm if both are set for some reason.

Anyway, if you want to keep it, make sure to add it to the model as well!

Reply:
Added constraint to model as well. I think this is needed, as both switches would get hidden if both flags are set in the db. Then the only way to recover would be to edit  values in the database by hand.

Reply:
Good point indeed!

Reply:
```python
def validate_is_always_visible(self, field):
    if self.is_always_visible.data and self.is_hidden.data:
        raise ValidationError('These two options cannot be used at the same time')

validate_is_hidden = validate_is_always_visible
```

Reply:
Done